src/Plugin/Plugins/TempHumidPlugin.h

```diff
@@ -0,0 +1,242 @@
+     *
+     * @param[in] slotInterf    Slot interface
+     */
+    void setSlot(const ISlotPlugin* slotInterf) final;
+
+    /**
+     * This method will be called in case the plugin is set active, which means
+     * it will be shown on the display in the next step.
+     *
+     * @param[in] gfx   Display graphics interface
+     */
+    void active(IGfx& gfx) final;
+
+    /**
+     * Update the display.
+     * The scheduler will call this method periodically.
+     *
+     * @param[in] gfx   Display graphics interface
+     */
+    void update(IGfx& gfx) final;
+
+    /**
+     * Start the plugin.
+     * Overwrite it if your plugin needs to know that it was installed.
+     */
+    void start() final;
+
+    /**
+     * Process the plugin.
+     * Overwrite it if your plugin has cyclic stuff to do without being in a
+     * active slot.
+     */
+    void process(void) final;    
+
+    /**
+     * Protect against concurrent access.
+     */
+    void lock(void) const;
+
+    /**
+     * Unprotect against concurrent access.
+     */
+    void unlock(void) const;
+
+private:
+    /**
+     * Icon width in pixels.
+     */
+    static const int16_t        ICON_WIDTH          = 8;
+
+    /**
+     * Icon height in pixels.
+     */
+    static const int16_t        ICON_HEIGHT         = 8;
+
+    /**
+     * Path to the temperature icon
+     */    
+    static const char*          IMAGE_PATH_TEMP_ICON;
+
+    /**
+     * Path to the humidity icon
+     */
+    static const char*          IMAGE_PATH_HUMID_ICON;
+
+    /**
+     * Read sensor only every N milliseconds (currently 90 seconds)
+     */
+    static const uint32_t       SENSOR_UPDATE_PERIOD = (90U * 1000U);
+
+    Canvas*                     m_textCanvas;               /**< Canvas used for the text widget. */
+    Canvas*                     m_iconCanvas;               /**< Canvas used for the bitmap widget. */
+    BitmapWidget                m_bitmapWidget;             /**< Bitmap widget, used to show the icon. */
+    TextWidget                  m_textWidget;               /**< Text widget, used for showing the text. */
+    String                      m_text;
+
+    uint8_t                     m_page;                     /**< Number of page, which to show */
+    unsigned long               m_pageTime;                 /**< How long to show page (1/4 slot-time or 10s default) */    
+    SimpleTimer                 m_timer;                    /**< Timer for changing page */
+    SemaphoreHandle_t           m_xMutex;                   /**< Mutex to protect against concurrent access. */
+
+    DHTesp                      m_dht;                      /**< sensor object*/
+    float                       m_humid;                    /**< last sensor humidity value*/
+    float                       m_temp;                     /**< last sensor temperature value*/
+    unsigned long               m_last;                     /**< last sensor read timestamp*/
```

Comment:
Please start in all member descriptions uppercase to be consistent.
Please add to all description the unit of the value.
Please add additional space at the comment end.